lib/sidekiq_alive/worker.rb

```diff
@@ -1,22 +1,28 @@
 # frozen_string_literal: true
 
+require "timeout"
+
 module SidekiqAlive
   class Worker
     include Sidekiq::Worker
     sidekiq_options retry: true
 
+    TIMEOUT = 30
+
     # Passing the hostname argument it's only for debugging enqueued jobs
     def perform(_hostname = SidekiqAlive.hostname)
-      # Checks if custom liveness probe passes should fail or return false
-      unless config.custom_liveness_probe.call
-        Rails.logger.error("SidekiqAlive: failed liveness probe, returning")
-        return
-      end
+      Timmeout.timeout(TIMEOUT) do
+        # Checks if custom liveness probe passes should fail or return false
+        unless config.custom_liveness_probe.call
+          Rails.logger.error("SidekiqAlive: failed liveness probe, returning")
```

Comment:
I think a more appropriate message would be "SidekiqAlive: failed custom liveness probe, returning" here